### ISSUE TITLE
Conda crashes with `IndexError: tuple index out of range`

### ISSUE THREAD

**Issue by SylvainCorlay** [conda/conda#8970]
We are getting reports about conda crashing upon installation of xeus-cling.

https://github.com/QuantStack/xeus-cling/issues/254

```
(base) root@ubuntu_18-04_svr_rsh:~# conda install -c conda-forge xeus-cling
Collecting package metadata (repodata.json): done
Solving environment: failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/exceptions.py", line 1062, in __call__
        return func(*args, **kwargs)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/cli/main_install.py", line 20, in execute
        install(args, parser, 'install')
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/cli/install.py", line 256, in install
        force_reinstall=context.force_reinstall or context.force,
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 112, in solve_for_transaction
        force_remove, force_reinstall)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 150, in solve_for_diff
        force_remove)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 262, in solve_final_state
        ssc = self._run_sat(ssc)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 780, in _run_sat
        repodata_fn=self._repodata_fn)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 1220, in solve
        self.find_conflicts(specs, specs_to_add, history_specs)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 336, in find_conflicts
        bad_deps = self.build_conflict_map(specs, specs_to_add, history_specs)
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 418, in build_conflict_map
        bad_deps.extend(g.depth_first_search_by_name(spec, dep, sdeps[spec]))
      File "/opt/anaconda3/lib/python3.7/site-packages/conda/models/prefix_graph.py", line 450, in depth_first_search_by_name
        matchspecs_for_chain.append(MatchSpec.merge(matchspecs_to_merge)[0])
    IndexError: tuple index out of range

`$ /opt/anaconda3/bin/conda install -c conda-forge xeus-cling`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/opt/anaconda3/bin/conda
             CONDA_PREFIX=/opt/anaconda3
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/opt/anaconda3/bin/python
               CONDA_ROOT=/opt/anaconda3
              CONDA_SHLVL=1
     JAVA_LD_LIBRARY_PATH=/opt/anaconda3/jre/lib/amd64/server
                     PATH=/opt/anaconda3/bin:/opt/anaconda3/condabin:/usr/local/sbin:/usr/local/
                          bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bi
                          n
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : base
    active env location : /opt/anaconda3
            shell level : 1
       user config file : /root/.condarc
 populated config files :
          conda version : 4.7.9
    conda-build version : 3.17.8
         python version : 3.7.3.final.0
       virtual packages :
       base environment : /opt/anaconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/anaconda3/pkgs
                          /root/.conda/pkgs
       envs directories : /opt/anaconda3/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.9 requests/2.21.0 CPython/3.7.3 Linux/4.15.0-54-generic ubuntu/18.04.2 glibc/2.27
                UID:GID : 0:0
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.

If submitted, this report will be used by core maintainers to improve
future releases of conda.
Would you like conda to send this report to the core maintainers?

[y/N]: y
Upload successful.

Thank you for helping to improve conda.
Opt-in to always sending reports (and not see this message again)
by running

    $ conda config --set report_errors true
```

**Comment by ericdill** [comment]
What's currently in your environment? Can you show the output of `conda list --show-channel-urls`? I was not successful at reproducing this inside the continuumio/miniconda3 docker container

```
$ docker run -it continuumio/miniconda3 /bin/bash
```
Then, inside the docker container:
```
(base) root@29416fc31b61:/# conda install conda=4.7.9 -y
<snip>
(base) root@29416fc31b61:/# conda install conda-build=3.17.8 -y
<snip>
(base) root@29416fc31b61:/# conda install -c conda-forge xeus-cling -y
<snip>
(base) root@29416fc31b61:/# conda list | grep xeus                        
xeus                      0.21.1               h4d8c418_2    conda-forge  
xeus-cling                0.7.1                he513fc3_0    conda-forge  

```

**Comment by msarahan** [comment]
This is fixed in conda 4.7.10.

**Comment by SylvainCorlay** [comment]
> What's currently in your environment? Can you show the output of conda list --show-channel-urls? I was not successful at reproducing this inside the continuumio/miniconda3 docker container

Yes, this occcured in a *clean* anaconda, not miniconda. We recommended using miniconda instead. The original poster said that this was not possible for them due to external constrains.

> This is fixed in conda 4.7.10.

Thanks!
